sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1340,12 +1344,12 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::ExtractOrCreateCurlCo
 
   if (options.ConnectionTimeout != CurlTransportOptions::DefaultConnectionTimeout)
   {
-    if (!SetLibcurlOption(newHandle, CURLOPT_CONNECTTIMEOUT, options.ConnectionTimeout, &result))
+    if (!SetLibcurlOption(newHandle, CURLOPT_CONNECTTIMEOUT_MS, options.ConnectionTimeout, &result))
```

Comment:
nit: The comment `// using DefaultConnectionTimeout or 0 result in the same setting` probably belongs here as well.